homeassistant/components/wallbox/__init__.py

```diff
@@ -0,0 +1,112 @@
+"""The Wallbox integration."""
+import asyncio
+import logging
+
+import requests
+from wallbox import Wallbox
+
+from homeassistant import exceptions
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_CONNECTIONS, CONF_STATION, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["sensor"]
+
+
+class WallboxHub:
+    """Wallbox Hub class."""
+
+    def __init__(self, station, username, password):
+        """Initialize."""
+        self._station = station
+        self._username = username
+        self._password = password
+
+    async def async_authenticate(self, hass) -> bool:
+        """Authenticate using Wallbox API."""
+        try:
+            wallbox = Wallbox(self._username, self._password)
+            await hass.async_add_executor_job(wallbox.authenticate)
+            return True
+        except requests.exceptions.HTTPError as wallbox_connection_error:
+            if wallbox_connection_error.response.status_code == 403:
+                raise InvalidAuth from wallbox_connection_error
+            raise ConnectionError from wallbox_connection_error
+
+    async def async_get_data(self, hass) -> bool:
+        """Get new sensor data for Wallbox component."""
+        try:
+            wallbox = Wallbox(self._username, self._password)
```

Comment:
Again, I do not fully understand. But let me explain my logic here. I want people to be able to create multiple instances of this platform/integration, because they might have multiple stations which might use multiple logins (a few people on the forum already mention using different stations). So, for me a wallbox is the combination of Station/User/Password. In setup_entry I instantiate an instance of WallboxHub and pass it in hass.data[DOMAIN][CONF_CONNECTIONS][entry.entry_id].

Reply:
If they create multiple instances of your platform async_setup_entry will be called multiple tines, meaning you have 1 instance of WallBoxHub for each username/password. So it would be natural to store the library object WallBox in WallBoxHub.

Reply:
When working on your previous comment I actually got to the same conclusion (ended up storing wallbox in a hub variable) and I think that is exactly what you mean here. So, good suggestion and done.